instant-xml-macros/src/lib.rs

```diff
@@ -47,18 +52,49 @@ impl ContainerMeta {
                         "attribute key invalid in container xml attribute",
                     ))
                 }
-                MetaItem::Ns(ns) => meta.ns = ns,
-                MetaItem::Rename(lit) => meta.rename = Some(lit),
+                MetaItem::Ns(namespace) => ns = namespace,
+                MetaItem::Rename(lit) => rename = Some(lit),
                 MetaItem::RenameAll(lit) => {
-                    meta.rename_all = match RenameRule::from_str(&lit.to_string()) {
+                    rename_all = match RenameRule::from_str(&lit.to_string()) {
                         Ok(rule) => rule,
                         Err(err) => return Err(syn::Error::new(span, err)),
                     };
                 }
-                MetaItem::Scalar => meta.scalar = true,
+                MetaItem::Scalar => scalar = true,
             }
         }
-        Ok(meta)
+
+        Ok(Self {
+            input,
+            ns,
+            rename,
+            rename_all,
+            scalar,
+        })
+    }
+
+    fn lifetimed_generics(&self) -> Generics {
```

Comment:
Using lifetime as a verb feels a little odd. Maybe just call it `xml_generics()`?